packages/astro/src/assets/build/generate.ts

```diff
@@ -103,9 +104,13 @@ export async function generateImage(
 
 	// If the image is local, we can just read it directly, otherwise we need to download it
 	if (isLocalImage) {
+		const outDir = isServerLikeOutput(config)
+			? config.build.server
+			: getOutDirWithinCwd(config.outDir);
```

Comment:
It might be better to use `getOutDirWithinCwd()` on this line instead:

https://github.com/withastro/astro/blob/2365c124645d5067a12987f205cee23a45d1d13d/packages/astro/src/assets/build/generate.ts#L51